src/Chunk.ts

```diff
@@ -403,10 +403,10 @@ export default class Chunk {
 	}
 
 	private prepareDynamicImports({ format }: OutputOptions) {
-		const es = format === 'es';
+		const esm = format === 'esm' || format === 'es';
```

Comment:
Ideally all these checks should only check one format string - either `esm` or `es`, with internal rewriting to ensure only one is used.

Reply:
I guess this merge should happen in `mergeOptions`?